activation/handler_v2.go

```diff
@@ -405,11 +428,124 @@ func (h *HandlerV2) validatePositioningAtx(publish types.EpochID, golden, positi
 	return posAtx.TickHeight(), nil
 }
 
+// Validate marriages and return married IDs.
+// Note: The order of returned IDs is important and must match the order of the marriage certificates.
+// The MarriageIndex in PoST proof matches the index in this marriage slice.
+func (h *HandlerV2) validateMarriages(atx *wire.ActivationTxV2) ([]types.NodeID, error) {
+	if len(atx.Marriages) == 0 {
+		return nil, nil
+	}
+	var marryingIDs []types.NodeID
+	for i, m := range atx.Marriages {
+		var id types.NodeID
+		if m.ReferenceAtx == types.EmptyATXID {
+			id = atx.SmesherID
+		} else {
```

Comment:
Actually, I think we should check if all marriage certificates belong to distinct IDs.

>the main identity could include multiple posts of themselves?

That wouldn't be possible because we associate 1 marriage idx with every ID. Anyway, I will add an extra check.